Fix for PR 283 (#306)
rancher/fleet-product-docs: community-docs/next/modules/ROOT/pages/explanations/namespaces.adoc

created bundles from git by putting labels in the `fleet.yaml` file or on the
`metadata.labels` field on the `GitRepo`.

[#_restricting_gitrepos]

> === Restricting GitRepos

Reviewer: We may want to mention the new `Policy` resource here, recommending that users rely on them instead of `GitRepoRestrictions`, which will be deprecated in a future release.

A namespace can contain multiple `GitRepoRestriction` resources. All `GitRepos`
created in that namespace will be checked against the list of restrictions. If a
`GitRepo` violates one of the constraints its `BundleDeployment` will be in an
error state and won't be deployed.

This can also be used to set the defaults for GitRepo's `serviceAccount` and
`clientSecretName` fields.

[,yaml]
----
kind: GitRepoRestriction
apiVersion: fleet.cattle.io/v1alpha1
metadata:
  name: restriction
  namespace: typically-unique
allowedClientSecretNames: []
allowedRepoPatterns: []
allowedServiceAccounts: []
allowedTargetNamespaces: []
defaultClientSecretName: ""
defaultServiceAccount: ""
----

==== Allowed Target Namespaces

This can be used to limit a deployment to a set of namespaces on a downstream

> cluster. If an allowedTargetNamespaces restriction is present, all `GitRepos`

Reviewer: ```suggestion
cluster. If an `allowedTargetNamespaces` restriction is present, all `GitRepos`
```

must specify a `targetNamespace` and the specified namespace must be in the

> allow list. This also prevents the creation of cluster wide resources.

Reviewer: nit: I _think_ this is how it should be written, but not entirely sure.
```suggestion
allow list. This also prevents the creation of cluster-wide resources.
```

==== Allowed target namespace selector

`allowedTargetNamespaceSelector` restricts deployments to namespaces on

> downstream clusters whose labels match the given selector. When this field is

Reviewer: nit: This could be read as if the matching labels would be the cluster's labels, as opposed to the namespace's labels. We could rephrase this as something like:
```suggestion
`allowedTargetNamespaceSelector` restricts deployments, on downstream clusters, to namespaces
with labels matching the given selector. When this field is
```

(edit: downgraded this to a nitpick, as you've clearly explained it below)

set in a `GitRepoRestriction`, all `GitRepos` must specify a `targetNamespace`.
The Fleet agent checks the labels of that namespace on the downstream cluster
before deploying a bundle:

* If the namespace exists and its labels match the selector, deployment proceeds.
* If the namespace exists but its labels do not match, the `BundleDeployment` is

> * set to an error state and the bundle is not deployed.

Reviewer: Unwanted list item:
```suggestion
set to an error state and the bundle is not deployed.
```
